### ISSUE TITLE
Quite long start times?

### ISSUE THREAD

**Issue by dbolser-ebi** [dbcli/mycli#465]
Perhaps I'm spoiled, but these start times seem quite long:
https://asciinema.org/a/jX5lJQ3Vr3Ax3BOELsdTcxLeb

**Comment by tsroten** [comment]
@dbolser-ebi Hello! Those do seem a bit long. To compare, I'm hitting anywhere from .354 to .510 seconds depending on which server I connect to. That includes opening an SSH connection to tunnel over.

Have you compared the start times on the same server with mysql's?

Also, for your credentials script, is it pulling the credentials from some kind of auth manager/encrypted file?

**Comment by dbolser-ebi** [comment]
https://asciinema.org/a/2vRFHWtxpZwBsivLPyWqngSon

No, but having the credentials script work with mysql's encrypted... thing is on my todo list.

**Comment by tsroten** [comment]
Certainly a big difference. I think it's worth exploring what some of the bottlenecks might be. Especially for the `-e` argument, e.g. not setting up the completion engine.

**Comment by amjith** [comment]
Agreed! We need to run a profiler and figure out where we're spending time. I've noticed this on mycli and pgcli as well the startup times are getting worse.

**Comment by dbolser-ebi** [comment]
I think the native client only sets up autocomplete for the given database
(or nothing if one isn't selected). I'm guessing mycli sets up all metadata
for all databases?

There are quite a few DBs on these two servers, each with 50+ tables.

BTW, the -e is just to get clean timing, I don't consider this a bug
per-se, although obviously this could be 'fixed'.

​

**Comment by amjith** [comment]
We also collect completion data for the given database. We refresh
completion when you switch to a different database.

**Comment by rolandwalker** [comment]
Closing since this is a bit vague to be actionable.  Anyone should feel free to reopen with specifics.
